ipu-plugin/pkg/ipuplugin/vendorservice.go

```diff
@@ -0,0 +1,233 @@
+	ACC_VPORT_ID  = "02"
+
+	DEVICE = "0x1452"
+	VENDOR = "0x8086"
+
+	ACC_IP  = "192.168.11.2"
+	HOST_IP = "192.168.11.1"
+	PORT    = 50151
+)
+
+type NetworkHandler interface {
+	AddrAdd(link netlink.Link, addr *netlink.Addr) error
+	AddrList(link netlink.Link, family int) ([]netlink.Addr, error)
+	LinkList() ([]netlink.Link, error)
+}
+
+type NetworkHandlerImpl struct{}
+
+func (h *NetworkHandlerImpl) AddrAdd(link netlink.Link, addr *netlink.Addr) error {
+	return netlink.AddrAdd(link, addr)
+}
+func (h *NetworkHandlerImpl) AddrList(link netlink.Link, family int) ([]netlink.Addr, error) {
+	return netlink.AddrList(link, family)
+}
+func (h *NetworkHandlerImpl) LinkList() ([]netlink.Link, error) {
+	return netlink.LinkList()
+}
+
+type FileSystemHandler interface {
+	GetDevice(iface string) ([]byte, error)
+	GetVendor(iface string) ([]byte, error)
+}
+type FileSystemHandlerImpl struct{}
+
+func (fs *FileSystemHandlerImpl) GetDevice(iface string) ([]byte, error) {
+	return os.ReadFile(fmt.Sprintf("/sys/class/net/%s/device/device", iface))
+}
+func (fs *FileSystemHandlerImpl) GetVendor(iface string) ([]byte, error) {
+	return os.ReadFile(fmt.Sprintf("/sys/class/net/%s/device/vendor", iface))
+}
+
+var fileSystemHandler FileSystemHandler
+var networkHandler NetworkHandler
+
+func initHandlers() {
+	if fileSystemHandler == nil {
+		fileSystemHandler = &FileSystemHandlerImpl{}
+	}
+	if networkHandler == nil {
+		networkHandler = &NetworkHandlerImpl{}
+	}
+}
+
+func isPF(iface string) bool {
+	device, _ := fileSystemHandler.GetDevice(iface)
+	vendor, _ := fileSystemHandler.GetVendor(iface)
+	return strings.TrimSpace(string(device)) == DEVICE && strings.TrimSpace(string(vendor)) == VENDOR
+}
+
+func getCommPf(isACC bool, linkList []netlink.Link) (netlink.Link, error) {
+	var pf netlink.Link
+	for i := 0; i < len(linkList); i++ {
+		mac := linkList[i].Attrs().HardwareAddr.String()
+		octets := strings.Split(mac, ":")
+
+		// Check the 4th octet which is used to identify the PF
+		if isACC && octets[3] == ACC_VPORT_ID {
+
+			// On ACC, the 4th octet in the base mac address may already be set to ACC_VPORT_ID
+			// The base mac address is set to be used by the PF mapped to the IMC, and has an IP address.
+			// Two distinguish between the two, we select the one which doesn't have an IP set already.
+			if list, _ := networkHandler.AddrList(linkList[i], netlink.FAMILY_V4); len(list) == 0 {
+				pf = linkList[i]
+				break
+			}
+
+		}
+
+		// Check the 4th octet which is used to identify the PF
+		if !isACC && octets[3] == HOST_VPORT_ID {
+			pf = linkList[i]
+			break
+		}
+	}
+
+	if pf == nil {
+		return nil, fmt.Errorf("unable to identify any PFs. Check that the idpf driver is available")
+	}
+
+	return pf, nil
+}
+
+func setIP(link netlink.Link, ip string) error {
+	list, err := networkHandler.AddrList(link, netlink.FAMILY_V4)
+
+	if err != nil {
+		return fmt.Errorf("unable to get the ip address of link: %v", err)
+	}
+
+	if len(list) == 0 {
```

Comment:
think this answers my question - it's implied list will be 0 and err would be nil